### PR TITLE
Add pystreamer

### PR DESCRIPTION
**Pull request by Arnoldosmium**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/pystreamer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  build: {}
+  host:
+    - python >=2.7
+    - pip
+  run:
+    - python >=2.7
+  test:
```
Comment (volpatto) [inline recipes/pystreamer/meta.yaml]: I think this is an invalid subsection to requirements. Try to put a `requires` subsection in `test`.

Comment (Arnoldosmium) [reply recipes/pystreamer/meta.yaml]: thanks a lot for the detailed instruction! just realize the test modules are not even packaged so i'll omit the test section until the next release :)

File: recipes/pystreamer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 200e910b273570aaf93aec076a399dfec413afa6bfc049943a0f2599c22fa937
+  
+build:
+  noarch: python
+  number: 0
+  script: "PACKAGE_VERSION={{ version }} {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+
+requirements:
+  build: {}
```
Comment (xhochy) [inline recipes/pystreamer/meta.yaml]: Omit this
```suggestion
```

File: recipes/pystreamer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  imports:
+    - streamer
+
+about:
+  home: https://github.com/Arnoldosmium/pystreamer
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'A chain-able generator for python; inspired by the syntactical sugar of java stream'
+  doc_url: ''
```
Comment (xhochy) [inline recipes/pystreamer/meta.yaml]: Omit empty sections
```suggestion
```

File: recipes/pystreamer/meta.yaml (hunk 2 above)
Comment (Arnoldosmium) [reply recipes/pystreamer/meta.yaml]: Thanks for pinpointing out!

File: recipes/pystreamer/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{% set name = "pystreamer" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  # url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  url: https://github.com/Arnoldosmium/pystreamer/archive/{{ version }}.tar.gz
```
Comment (xhochy) [inline recipes/pystreamer/meta.yaml]: Why the switch to github? We normally prefer `pypi.org` as the main source.

File: recipes/pystreamer/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  sha256: e9ff94c97517a686d41e58fa163e543a340bf802b9c5a911aa7c4833c05c29e4
+  
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+
+requirements:
+  host:
+    - python >=2.7
```
Comment (marcelotrevisani) [inline recipes/pystreamer/meta.yaml]: the ``python >=2.7`` is not necessary, please remove it

File: recipes/pystreamer/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    - pip
+  run:
+    - python >=2.7
```
Comment (marcelotrevisani) [inline recipes/pystreamer/meta.yaml]: also here

File: recipes/pystreamer/meta.yaml (hunk 5 above)
Comment (Arnoldosmium) [reply recipes/pystreamer/meta.yaml]: I'd like to restrict my package to python 2.7 and python 3.x only. No versions <=2.6

Comment (xhochy) [reply recipes/pystreamer/meta.yaml]: Conda-forge only supports 2.7, 3.6 and 3.7, this selector is already the global default and thus redundant 😉

**Pull request merged**